src/AbstractRole.php

```diff
@@ -103,16 +103,33 @@ public function addChild($child)
      */
     public function setParent($parent)
     {
-        $this->parent = $parent;
-
+        if (null === $this->parents) {
+            $this->parents = [];
+        }
+        if (!in_array($parent, $this->parents)) {
+            $this->parents[] = $parent;
+        }
         return $this;
     }
 
     /**
-     * @return null|RoleInterface
+     * @return null|RoleInterface|array
      */
     public function getParent()
     {
-        return $this->parent;
+        if (1 === count($this->parents)) {
```

Comment:
We also need a check for a `null` value, do we not?

Reply:
If `$this->parents` is null count() will returns 0.

Reply:
@ezimuel  Not on PHP 7.2+, see: https://3v4l.org/WpsN8

Reply:
I mean, yes, but we get also warning ;-)

Reply:
@webimpress you right, I need to update the PHP documentation :)